sapi/cgi/fastcgi.c

```diff
@@ -680,7 +680,9 @@ int fcgi_listen(const char *path, int backlog)
 #endif
 	    bind(listen_socket, (struct sockaddr *) &sa, sock_len) < 0 ||
 	    listen(listen_socket, backlog) < 0) {
-
```

Comment:
Maybe you'd better split the PR, this will makes review easier. More detailed explanation will be appreciated, and if possible maybe add some tests?